json_config_expander/config_expander.py

```diff
@@ -0,0 +1,49 @@
+"""
+ConfigExpander class makes all work
+"""
+from copy import deepcopy, copy
+from itertools import product, chain
+
+
+def seq_iter(obj):
+	# iterate dict and list in same way
+	return obj if isinstance(obj, dict) else range(len(obj))
+
+
+class ConfigExpander:
+	def __init__(self, expand_char='*'):
+		self.expand_char = expand_char
+
+	def run_on_each_config(self, base_config, function):
+		expanded_configs = self.expand_configs(base_config)
+		return [function(deepcopy(config)) for config in expanded_configs]
+
+	def expand_configs(self, base_config):
+		"""
+		For each child we calculate its versions recursively and create new config with all possible combinations
+		"""
+		if not isinstance(base_config, dict) and not isinstance(base_config, list):
+			return [base_config]
+		expanded_config = self._expand_config_children(base_config)
+		return self._create_cartesian_product_from_children(expanded_config)
+
+	def _expand_config_children(self, config):
+		expanded_config = copy(config)
+		for key in seq_iter(expanded_config):
```

Comment:
I don't like the usage of seq_iter here. Although its cool and elegant, it resolves in a confusing code, as I don't expect key to be an index of an array on the first read. It also might resolve with changing the collection while iterating it if the implementation of seq_iter() or the internal functions it uses will change. IMO It will be better to separate the handling of a dict and an array, and in both of them create a new collection (using comprehension maybe) instead of changing the current one.